File: mobile-ios/OffMe/OffMeApp.swift
import SwiftUI
import UIKit

class AppDelegate: NSObject, UIApplicationDelegate {
    func application(
        _ application: UIApplication,
        didRegisterForRemoteNotificationsWithDeviceToken deviceToken: Data
    ) {
        Task { @MainActor in
            PushNotificationService.shared.handleDeviceToken(deviceToken)
        }
    }
}

@main
struct OffMeApp: App {
    @UIApplicationDelegateAdaptor(AppDelegate.self) var appDelegate
    @StateObject private var auth = AuthStore()

    var body: some Scene {
        WindowGroup {
            RootView()
                .environmentObject(auth)
                .preferredColorScheme(.light)
                .onAppear {
                    PushNotificationService.shared.registerIfNeeded()
                }
                .task(id: auth.accessToken) {
                    if let token = auth.accessToken {
                        await PushNotificationService.shared.syncPendingToken(authToken: token)
                    }
                }
        }
    }
}

File: mobile-android/app/src/main/kotlin/com/offme/OffMeApp.kt
package com.offme

import android.app.Application
import com.offme.data.api.ApiClient
import com.offme.data.auth.AuthStore

class OffMeApp : Application() {
    lateinit var apiClient: ApiClient
        private set

    lateinit var authStore: AuthStore
        private set

    override fun onCreate() {
        super.onCreate()
        instance = this
        apiClient = ApiClient()
        authStore = AuthStore(apiClient, this)
    }

    companion object {
        lateinit var instance: OffMeApp
            private set
    }
}